test/util.js

```diff
@@ -107,27 +108,40 @@ exports.inBrowserContext = () => {
  * @param {string} relativePath Relative path within the test directory. For example "jsdom/files/test.html"
  * @returns {string} URL
  */
-function getTestFixtureUrl(relativePath) {
+exports.getTestFixtureUrl = relativePath => {
   /* globals location */
   if (exports.inBrowserContext()) {
     // location is a Location or WorkerLocation
     return location.origin + "/base/test" + (relativePath[0] === "/" ? "" : "/") + relativePath;
   }
-
   return toFileUrl(__dirname, relativePath);
-}
+};
 
 /**
  * Reads a static fixture file as utf8.
  * If running tests from node, the file will be read from the file system
  * If running tests using karma, a http request will be performed to retrieve the file using karma's server.
  * @param {string} relativePath Relative path within the test directory. For example "jsdom/files/test.html"
  */
-exports.readTestFixture = relativePath => {
+exports.readTestFixture = async relativePath => {
   if (exports.inBrowserContext()) {
-    return request(getTestFixtureUrl(relativePath), { timeout: 5000 });
+    const abortController = new AbortController();
+    const { signal } = abortController;
+    const timeout = setTimeout(() => {
+      abortController.abort();
+    }, 5000);
+    try {
+      const response = await fetch(exports.getTestFixtureUrl(relativePath), { method: "GET", signal });
```

Comment:
It looks like we only use node-fetch in a browser context, right? In that case you could replace this with `await self.fetch(...)` and the above with `new self.AbortController()`, and remove the dependencies, I am pretty sure. Do you mind trying that?

Reply:
So, `self` is the stand-in for the `window` object here?

Reply:
Actually, there is one spot within the Node execution context where node-fetch is required, the script `start-wpt-server.js`. Still makes it a dev-dependency only, so that should be acceptable, right?

Reply:
Yeah, seems reasonable. Maybe that could be replaced too but it's not necessarily worth doing right now.